src/server/system_services/bucket_server.js

```diff
@@ -1900,6 +1905,112 @@ function validate_non_nsfs_bucket_creation(req) {
     }
 }
 
+async function put_bucket_replication(req) {
+    dbg.log0('put_bucket_replication:', req.rpc_params);
+    const bucket = find_bucket(req);
+
+    validate_replication(req, bucket.name.unwrap(), true);
+    const replication_rules = normalize_replication(req);
+
+    const bucket_replication_id = bucket.replication_policy_id;
+    const replication_id = bucket_replication_id ?
+        await replication_store.instance().update_replication(replication_rules, bucket_replication_id) :
+        await replication_store.instance().insert_replication(replication_rules);
+
+    console.log('update_bucket_class_replication: replication_id: ', replication_id);
+
+    await system_store.make_changes({
+        update: {
+            buckets: [{ _id: bucket._id, replication_policy_id: replication_id }]
+        }
+    });
+}
+
+async function get_bucket_replication(req) {
+    dbg.log0('get_bucket_replication:', req.rpc_params);
+    const bucket = find_bucket(req);
+
+    const replication_id = bucket.replication_policy_id;
+    if (!replication_id) throw new RpcError('NO_REPLICATION_ON_BUCKET');
+
+    const replication = await replication_store.instance().get_replication_by_id(replication_id);
+    const bucket_names_replication = _.map(replication, rule =>
+        ({ ...rule, destination_bucket: system_store.data.get_by_id(rule.destination_bucket).name }));
+
+    return bucket_names_replication;
+}
+
+async function delete_bucket_replication(req) {
+    dbg.log0('delete_bucket_replication:', req.rpc_params);
+    const bucket = find_bucket(req);
+    const replication_id = bucket.replication_policy_id;
+    if (!replication_id) return;
+    await system_store.make_changes({
+        update: {
+            buckets: [{
+                _id: bucket._id,
+                $unset: { replication_policy_id: 1 }
+            }]
+        }
+    });
+
+    // delete replication from replication collection
+    await replication_store.instance().delete_replication_by_id(replication_id);
+}
+
+function validate_replication(req, bucket_name, min_required) {
+    const replication_rules = req.rpc_params.replication_policy;
+    // num of rules in configuration must be in defined limits
+    if (replication_rules.length > config.BUCKET_REPLICATION_MAX_RULES ||
+        (min_required && replication_rules.length < 1)) throw new RpcError('INVALID_REPLICATION_POLICY');
+
+    let rule_ids = [];
+    let pref_by_dst_bucket = {};
+
+    _.map(replication_rules, rule => {
```

Comment:
you don't need to use `_.map`. just use for-of 
```suggestion
    for (const rule of replication_rules) {
```